### PR TITLE
add firmlink to support mounting /cvmfs in Catalina osx 10.15

### PR DESCRIPTION
**Pull request by siscia**
I also removed the `sudo` that should not be necessary since we are running as sudo anyway.

### CODE REVIEW

File: cvmfs/cvmfs_config [hunk 1]
```diff
@@ -1833,15 +1833,23 @@ EOF
...
+     # firmlinks from `/cvmfs` to `/Users/shared/cvmfs`
      if sw_vers | grep 10.15; then
-        sudo mkdir -p /Users/shared/cvmfs
-        sudo chown cvmfs:cvmfs /Users/shared/cvmfs
-        sudo chmod 777 /Users/shared/cvmfs
+        mkdir -p /Users/shared/cvmfs
+        chown cvmfs:cvmfs /Users/shared/cvmfs
+        chmod 777 /Users/shared/cvmfs
+        if ! grep -q "cvmfs" /etc/synthetic.conf 2> /dev/null ; then
+          echo -e "# Autogenerated by CernVM-FS, do not delete nor edit" >> /etc/synthetic.conf
```
Comment (jblomer) [inline cvmfs/cvmfs_config]: I think we cannot do that.  There might be other parties interested in editing the _file_. We can only set the "auto-generated" marker for our _line_, if that is possible (i.e., if lines can have trailing comments).  Or, we have to wrap the `cvmfs` line between begin and end comment lines.

File: cvmfs/cvmfs_config [hunk 2]
```diff
@@ -1833,15 +1833,23 @@ EOF
...
+     # In order to keep the standard CVMFS mountpoint, we decide to implement
+     # firmlinks from `/cvmfs` to `/Users/shared/cvmfs`
      if sw_vers | grep 10.15; then
-        sudo mkdir -p /Users/shared/cvmfs
-        sudo chown cvmfs:cvmfs /Users/shared/cvmfs
-        sudo chmod 777 /Users/shared/cvmfs
+        mkdir -p /Users/shared/cvmfs
+        chown cvmfs:cvmfs /Users/shared/cvmfs
+        chmod 777 /Users/shared/cvmfs
+        if ! grep -q "cvmfs" /etc/synthetic.conf 2> /dev/null ; then
```
Comment (jblomer) [inline cvmfs/cvmfs_config]: Perhaps better `grep -q "^cvmfs\t"` to be more precise

File: cvmfs/cvmfs_config (hunk 1 above)
Comment (siscia) [reply cvmfs/cvmfs_config]: Aw I meant that the line below was autogenerated, not the whole file.

Something like this:

```
# The line below was autogenerated by CernVM-FS, do not delete nor edit it
cvmfs /Users/Shared/cvmfs
```

File: cvmfs/cvmfs_config (hunk 2 above)
Comment (siscia) [reply cvmfs/cvmfs_config]: I intentionally leave it a little unprecise to let user play with it.

With the assumption that they may want to use something different, but that such something will contains `cvmfs` in the name.

I guess is not a big deal to have one more firmlink anyway!

Comment (jblomer) [reply cvmfs/cvmfs_config]: It's fine to let the user add more symlinks.  But in this way, if there is some "cvmfs stuff" present in the file, we won't add our version of the symlink.  And I think we should.

File: cvmfs/cvmfs_config (hunk 1 above)
Comment (siscia) [reply cvmfs/cvmfs_config]: The manual of `syntetich.conf` indicate that only the line that starts with `#` are comment, so we should not use trailing comments.

File: cvmfs/cvmfs_config (hunk 2 above)
Comment (siscia) [reply cvmfs/cvmfs_config]: Fixed!

File: cvmfs/cvmfs_config [hunk 3]
```diff
@@ -1833,15 +1833,23 @@ EOF
      # SIP) create directory in root `/`.
      # We decide to move the standard installation directory on
      # `/Users/shared/cvmfs` instead of `/cvmfs`
+     # In order to keep the standard CVMFS mountpoint, we decide to implement
+     # firmlinks from `/cvmfs` to `/Users/shared/cvmfs`
      if sw_vers | grep 10.15; then
```
Comment (jblomer) [inline cvmfs/cvmfs_config]: Ah, sorry, I forgot: this should check for >= 10.15, otherwise it will break again with 10.16...

Comment (siscia) [reply cvmfs/cvmfs_config]: Addressed in #2450

**Pull request merged**